teuthology/util/time.py

```diff
@@ -0,0 +1,41 @@
+import re
+
+from datetime import datetime, timedelta, timezone
+
+# When we're not using ISO format, we're using this
+TIMESTAMP_FMT = "%Y-%m-%d_%H:%M:%S"
+
+def parse_timestamp(timestamp: str) -> datetime:
+    """
+    timestamp: A string either in ISO format or TIMESTAMP_FMT.
+               If no timezone is specified, UTC is assumed.
+
+    :returns: a datetime object
+    """
+    try:
+        dt = datetime.fromisoformat(timestamp)
+    except ValueError:
+        dt = datetime.strptime(timestamp, TIMESTAMP_FMT)
+    if dt.tzinfo is None:
+        dt = dt.replace(tzinfo=timezone.utc)
+    return dt
+
+def parse_offset(offset: str) -> timedelta:
+    err_msg = "Offsets must either be an ISO-formatted timestamp or " \
```

Comment:
ISO-formatted timestamp, could please elaborate, though, I haven't seen in the code it is parsing ISO format handling, because we fail with ValueError exception if we do not match the relative value.

Reply:
And did you mean ISO 8601? As far as I understand it is something like `[HH[:MM[:SS[.mmm[uuu]]]]][+HH:MM]`.

Reply:
Btw, I only could found scans of older 2016 year standard (the recent is 2019) and I guess they use a duration
<img width="712" alt="image" src="https://github.com/user-attachments/assets/e2957af6-1cb6-4a10-9a83-a1d1f97e8450">